lib/internal/modules/esm/resolve.js

```diff
@@ -587,6 +591,7 @@ function packageExportsResolve(
     const resolveResult = resolvePackageTarget(
       packageJSONUrl, target, '', packageSubpath, base, false, false, false,
       conditions,
+      true,
```

Comment:
I know it's not new in this PR, but I'm not a big fan of bare boolean arguments like this where someone can't tell at a glance what the `true` or `false` means when looking at the callsite. For this, adding comments like... the following example would be helpful:

```
  const resolveResult = resolvePackageTarget(packageJSONUrl, target, '', packageSubpath, base,
     false /* pattern */,
     false /* internal */,
     false /* isPathMap */,
     conditions,
     true /* mustBeInternalTarget */);
```

But not blocking for this PR.

Reply:
I'm marking this as resolved but also might say to use inlay hints in your language server and this should generally be ok (often can be done just for literals)

Reply:
I agree with @jasnell on this.

Even if your editor can explain it sufficiently, code sometimes shows up in other places (stack traces, diffs, etc.) If an options object isn't ideal, then comments can do a lot to make the `true, true, false, true, etc` less mysterious when you're doing a bisect or trying to figure out an error.